### ISSUE TITLE
Not doing the sell order 

### ISSUE THREAD

**Issue by alexmanda2000** [fj317/PumpBot#97]
Hello. I have tried the bot, but unfortunately it didn`t created the sell order for me.
I have the following error: 
Coin pair: IDEX
Waiting for coin to buy...
Buy order has been made!
Processing sell order.
Traceback (most recent call last):

  File "....\GitHub\PumpBot\PumpBot.py", line 284, in <module>
    coinOrderInfo = order["fills"][0]

IndexError: list index out of range

**Comment by rayures** [comment]
have the same issue. 

think something with the wait on buy complete is faulty. 

--
````# waits until the buy order has been confirmed
print("Waiting for coin to buy...")
while not(orderCompleted):
    pass
# when order compelted reset to false for next order
orderCompleted = False````

**Comment by fj317** [comment]
I've seen this error before but I have no idea whats causing it.

> think something with the wait on buy complete is faulty.

This could be the problem, however I've tested this specific code and it works as intended when I've tested it so I'm not sure if it is the problem.

**Comment by luxleeeet** [comment]
What is your sell order set to? I am not to used with binance but maybe there is a limit on how high sell order you can put that is relying on the current price of the coin, try and lower your profitmargin and see what happens?

**Comment by rayures** [comment]
today the bot failed again, hanging on " Waiting for coin to buy..." 

buy orders was filled like 14 hours ago ;-)

i think i notice this since the websocket update. if i have some time this week i will do some tests. 

> What is your sell order set to? I am not to used with binance but maybe there is a limit on how high sell order you can put that is relying on the current price of the coin, try and lower your profitmargin and see what happens?

10% aka 110 in the config ;-)  (not using it for massive dump and pumps)

**Comment by alexmanda2000** [comment]
> 
> 
> What is your sell order set to? I am not to used with binance but maybe there is a limit on how high sell order you can put that is relying on the current price of the coin, try and lower your profitmargin and see what happens?

Maybe because i put the percentage of wallet to 100 and profit marign to 200 i think this could be the problem. If I will set the percentage of wallet to 75 I think it will fix it

**Comment by rayures** [comment]
@inaleva, nice one. i tried something similar; a custom timeout for testing. So far it always worked, therefore i concluded it had to be something going to fast. 

`
starttime = time.time()
while not(orderCompleted):
    pass
    if time.time() - starttime > 1:
        print("break after 1s")
        break'

**Comment by INaleva** [comment]
@rayures I did the following fix on my own bot, and it works very well

**Comment by rayures** [comment]
Yes, i added it to my fork, will test the sell order later. Much better when you know python and proper coding ;-)

I still have that buy order issue, bot does not get the buy order filled info, so it hangs. Will make a new issue for that.

**Comment by fj317** [comment]
I've merged your latest pull request @INaleva into the code base now. If you make any other changes/fixes please do make a pull request and I'll happily add them.

**Comment by fj317** [comment]
Problem should now be fixed, closing issue.
